tests/dragonfly/replication_test.py

```diff
@@ -1999,3 +1999,29 @@ async def send_setex():
     assert set(keys_master) == set(keys_replica)
 
     await disconnect_clients(c_master, *[c_replica])
+
+
+@pytest.mark.asyncio
+async def test_saving_replica(df_local_factory):
+    tmp_file_name = "".join(random.choices(string.ascii_letters, k=10))
+
+    master = df_local_factory.create(proactor_threads=1)
+    replica = df_local_factory.create(proactor_threads=1, dbfilename=f"dump_{tmp_file_name}")
+    df_local_factory.start_all([master, replica])
+
+    c_master = master.client()
+    c_replica = replica.client()
+
+    await c_master.execute_command("DEBUG POPULATE 100000 key 4048 RAND")
+    await c_replica.execute_command(f"REPLICAOF localhost {master.port}")
+    await wait_available_async(c_replica)
+
+    async def save_replica():
+        await c_replica.execute_command("save")
+
+    save_task = asyncio.create_task(save_replica())
+    await asyncio.sleep(0.1)  # let save start
+    await c_replica.execute_command("replicaof no one")
+    await save_task
```

Comment:
Once we assert `replicaof no one` returns `OK`  and doesn't crash, the test is over. We can kill the replica and cancel the save task (it'll be knocked out with an error either way)

Reply:
Maybe we can busy loop on `saving_metric > 0` to start quicker? Not to save time _here_, but to be able to use much less than 100k keys (we rarely use such large values like 100k in pytests + debug)

Reply:
I updated the test to use 10k keys and a busy loop. I do want to want to wait for the save to finish and not just exit after issue replicaof no one, this test will check that saving will finish after promotion to master and not only try to reproduce a specific crash we had